scylla-cql/src/frame/request/mod.rs

```diff
@@ -74,7 +77,28 @@ pub trait SerializableRequest {
 /// Not intended for driver's direct usage (as driver has no interest in deserialising CQL requests),
 /// but very useful for testing (e.g. asserting that the sent requests have proper parameters set).
 pub trait DeserializableRequest: SerializableRequest + Sized {
-    fn deserialize(buf: &mut &[u8]) -> Result<Self, ParseError>;
+    fn deserialize(buf: &mut &[u8]) -> Result<Self, RequestDeserializationError>;
+}
+
+/// An error type returned by [`DeserializableRequest::deserialize`].
+/// This is not intented for driver's direct usage. It's a testing utility,
+/// mainly used by `scylla-proxy` crate.
+#[derive(Debug, Error)]
+pub enum RequestDeserializationError {
+    #[error("Low level deser error: {0}")]
+    LowLevelDeserialization(#[from] LowLevelDeserializationError),
+    #[error("Io error: {0}")]
+    IoError(#[from] std::io::Error),
+    #[error("Specified flags are not recognised: {:02x}", flags)]
+    UnknownFlags { flags: u8 },
+    #[error("Named values in frame are currently unsupported")]
+    NamedValuesUnsupported,
+    #[error("Expected SerialConsistency, got regular Consistency: {0}")]
+    ExpectedSerialConsistency(Consistency),
+    #[error(transparent)]
+    BatchTypeParse(#[from] BatchTypeParseError),
+    #[error("Unexpected batch statement kind: {0}")]
+    UnexpectedBatchStatementKind(u8),
 }
```

Comment:
If it's not intended to be exposed to users, maybe let's make it `doc(hidden)` ?